src/App.tsx

```diff
@@ -1,24 +1,180 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React,
+{
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMesage } from './components/ErrorMesage';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
+import { fetchTodos, fetchAddTodo, remove } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 10777;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [error, setError] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [deleteTodoId, setDeleteTodoId] = useState(0);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const addTodo = (title: string) => {
+    if (!title.trim()) {
+      setError('Title can\'t be empty');
+      setTimeout(() => {
```

Comment:
you should to clear your timeout if the component was deleted
Move it to the ErrorMessage